validator/validators/basic/basic_validator.go

```diff
@@ -0,0 +1,192 @@
+	"go.uber.org/multierr"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/awsservice"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/common"
+	"github.com/aws/amazon-cloudwatch-agent-test/validator/models"
+)
+
+const metricErrorBound = 0.1
+
+type BasicValidator struct {
+	vConfig models.ValidateConfig
+}
+
+var _ models.ValidatorFactory = (*BasicValidator)(nil)
+
+func NewBasicValidator(vConfig models.ValidateConfig) models.ValidatorFactory {
+	return &BasicValidator{
+		vConfig: vConfig,
+	}
+}
+
+func (s *BasicValidator) GenerateLoad() error {
+	var (
+		multiErr              error
+		metricSendingInterval = time.Minute
+		dataRate              = s.vConfig.GetDataRate()
+		agentCollectionPeriod = s.vConfig.GetAgentCollectionPeriod()
+		agentConfigFilePath   = s.vConfig.GetCloudWatchAgentConfigPath()
+		receivers             = s.vConfig.GetPluginsConfig()
+	)
+
+	if err := common.StartLogWrite(agentConfigFilePath, agentCollectionPeriod, metricSendingInterval, dataRate); err != nil {
+		multiErr = multierr.Append(multiErr, err)
+	}
+
+	// Sending metrics based on the receivers; however, for scraping plugin  (e.g prometheus), we would need to scrape it instead of sending
+	for _, receiver := range receivers {
+		if err := common.StartSendingMetrics(receiver, agentCollectionPeriod, metricSendingInterval, dataRate); err != nil {
+			multiErr = multierr.Append(multiErr, err)
+		}
+	}
+
+	return multiErr
+}
+
+func (s *BasicValidator) CheckData(startTime, endTime time.Time) error {
+	var (
+		multiErr         error
+		ec2InstanceId    = awsservice.GetInstanceId()
+		metricNamespace  = s.vConfig.GetMetricNamespace()
+		validationMetric = s.vConfig.GetMetricValidation()
+		validationLog    = s.vConfig.GetLogValidation()
+	)
+
+	for _, metric := range validationMetric {
+		metricDimensions := []types.Dimension{
+			{
+				Name:  aws.String("InstanceId"),
+				Value: aws.String(ec2InstanceId),
+			},
+		}
+		for _, dimension := range metric.MetricDimension {
+			metricDimensions = append(metricDimensions, types.Dimension{
+				Name:  aws.String(dimension.Name),
+				Value: aws.String(dimension.Value),
+			})
+		}
+		err := s.ValidateMetric(metric.MetricName, metricNamespace, metricDimensions, metric.MetricValue, metric.MetricSampleCount, startTime, endTime)
+		if err != nil {
+			multiErr = multierr.Append(multiErr, err)
+		}
+	}
+
+	for _, log := range validationLog {
+		err := s.ValidateLogs(log.LogStream, log.LogValue, log.LogLines, startTime, endTime)
+		if err != nil {
+			multiErr = multierr.Append(multiErr, err)
+		}
+	}
+
+	return multiErr
+}
+
+func (s *BasicValidator) Cleanup() error {
+	var (
+		dataType      = s.vConfig.GetDataType()
+		ec2InstanceId = awsservice.GetInstanceId()
+	)
+	switch dataType {
+	case "logs":
+		awsservice.DeleteLogGroup(ec2InstanceId)
+	}
+
+	return nil
+}
+
+func (s *BasicValidator) ValidateLogs(logStream string, logLine string, numberOfLogLine int, startTime, endTime time.Time) error {
```

Comment:
Nit: startTime and endTime share a type, but logStream and logLine don't.
 ```go
func (s *BasicValidator) ValidateLogs(logStream, logLine string, numberOfLogLine int, startTime, endTime time.Time) error {
```

Reply:
Agree with you. Thanks for catching it, Andrew. Not sure the reason why I changed it for starttime and endtime but not logstream and logline.